### ISSUE TITLE
Nefit app does not react on "Bedrijfsmodus status

### ISSUE THREAD

**Issue by Platvoetje** [robertklep/nl.nefit.easy#3]
"Hello
I have seen that the thermostaat modus has been deleted. However when I now try with Bedrijfsmodus and Homey he does not react in a flow.

**Comment by robertklep** [comment]
Can you show me your flow?

Bedrijfsmodus/Operating Mode shouldn't be settable, it's a read-only status parameter.

**Comment by Platvoetje** [comment]
Hello

 

See below the flow.

 

It worked before but now not anymore.

**Comment by robertklep** [comment]
You cannot post images to Github over e-mail, you need to post them directly on the website: https://github.com/robertklep/nl.nefit.easy/issues/3

**Comment by Platvoetje** [comment]
![image](https://github.com/user-attachments/assets/22c5c369-b880-479a-ad1d-7e25e1bb6eb2)

**Comment by robertklep** [comment]
Nothing was changed regarding Bedrijfsmodus in any recent changes, so I'm not sure why this isn't working anymore.

And since I haven't had a Nefit Easy myself for a long time, I cannot debug this issue either.

**Comment by robertklep** [comment]
The only thing I can try is see if something appears in a diagnostics report.

You can create one from the Homey mobile app:
* go the the More tab
* Apps
* Nefit Easy
* press the Gear icon in the top right
* press "Create Diagnostics Report"

**Comment by Platvoetje** [comment]
Rapport nr
ed9419a5-4081-4a70-9fab-f34e84601072

Friso

Op wo 6 nov 2024 om 09:13 schreef Robert Klep ***@***.***>

**Comment by robertklep** [comment]
Thanks.

It looks like you installed the test version of the app, then went back to the normal version. That's causing a problem, because the app is trying to update the `thermostat_mode` capability that was removed in the test version of the app.

Can you try installing the test version again and see if the flow works again?

https://homey.app/en-nl/app/nl.nefit.easy/Nefit-Easy/test/

**Comment by Platvoetje** [comment]
Hallo
If I install the test version then it still not works as I cannot select
thermostat_mode

Friso
Op wo 6 nov 2024 om 09:29 schreef Robert Klep ***@***.***>

**Comment by robertklep** [comment]
Why do you need `thermostat_mode` when your flow is using Operating Mode/Bedrijfsmodus?

**Comment by Platvoetje** [comment]
Hallo

Omdat hij op bedrijfsmodus niet reageert in de flow en dacht dat de
thermostaat modus wel zou werken.

Friso

Op wo 6 nov 2024 om 12:01 schreef Robert Klep ***@***.***>

**Comment by robertklep** [comment]
Thermostaatmodus is iets heel anders en niet relevant.

Maar ik begrijp dus dat de testversie ook niet werkt. Kun je nog een keer een diagnostic report opsturen vanuit de testversie van de app?

**Comment by Platvoetje** [comment]
Hallo

Nr is
fe621468-090f-4aa6-b825-88bd075f16f4

Friso

Op wo 6 nov 2024 om 12:59 schreef Robert Klep ***@***.***>

**Comment by robertklep** [comment]
Geen foutmeldingen te zien, dus het uitlezen gaat goed. Wellicht dat er iets in het protocol van Nefit is veranderd waardoor het niet werkt, maar zoals ik al zei, ik heb geen Nefit meer en kan dus ook niet nagaan of dat het geval is.

Eventueel kun je me de logingegevens van jouw Easy naar me opsturen, bijvoorbeeld in het commentaar van zo'n diagnostic report (*niet* door ze in een reactie op dit bericht te zetten, want dit is allemaal openbaar), dan kan ik op jouw Easy kijken of ik iets kan vinden waardoor het niet (meer) werkt.

**Comment by MarcovBarneveld** [comment]
Hoi,
Ik heb hetzelfde probleem. Geen van de status in bedrijfsmode geeft een TRUE terug.
Het kan heel goed zijn dat dit al een tijd speelt, want dit gebruik ik alleen bij lage buiten temperaturen.
Kan het in inderdaad zijn dat de naamgeving is veranderd.
Als ik de variabele "modus" uitlees die behoort bij de ALS kaart "Bedrijfsmodus is veranderd", dan krijg ik "central heating", "off" en "hot water".
Misschien helpt dit.
Marco

**Comment by robertklep** [comment]
@MarcovBarneveld ah, nu begrijp ik het probleem!

De waarden die je ziet ("central heating", "off" en "hot water") zijn wel de waarden die intern gebruikt zouden moeten worden voor de "Bedrijfsmodus is XXX" kaart.

Athom heeft bij firmware v12.2.0, die onlangs is uitgekomen, wel het één en ander aangepast aan hoe "capabilities", zoals Bedrijfsmodus, werken, maar die veranderingen zouden er niet voor moeten zorgen dat deze kaart niet meer werkt.

En zoals ik al zei: ik heb zelf geen Nefit Easy meer dus kan dit niet debuggen (tenzij iemand z'n credentials met me wil delen zodat ik even met hun Easy kan meekijken).

Maar een workaround is om niet meer de "Bedrijfsmodus is XXX" kaart te gebruiken, maar een Logica kaart ("[modus] is gelijk aan ...").

**Comment by robertklep** [comment]
Ik heb nog even gecheckt met de credentials van iemand die ik ken, en ik kan het probleem niet reproduceren. De kaart "Bedrijfsmodus is XXX" werkt gewoon.

**Comment by MarcovBarneveld** [comment]
Als ik test krijg ik dit. Geen van de 3 statussen werkt.

![ScreenHunter_1317 Nov  12 09 21](https://github.com/user-attachments/assets/27af652a-7315-478e-9408-9c294b1c11ad)
![ScreenHunter_1319 Nov  12 09 27](https://github.com/user-attachments/assets/91097af1-50c8-4139-8472-1417be4d013b)
![ScreenHunter_1320 Nov  12 09 28](https://github.com/user-attachments/assets/8d878118-8530-4888-a74e-2fb5c949106a)

**Comment by robertklep** [comment]
Die kaarten kun je op die manier niet testen (de testfunctionaliteit van Homey is nogal beperkt wat dat betreft).

Gebruik je overigens bewust de "NIET" uitgang van de kaarten? Deze:
![image](https://github.com/user-attachments/assets/ad1b83ac-d8e8-4ca3-8508-e1dd224f4043)

**Comment by MarcovBarneveld** [comment]
Ja, in dit testgeval is dat bewust omdat ze altijd FALSE geven om vervolgens te loggen wat er dan wel in de variabelen staat. 👍
